pkg/flow/internal/controller/component_registry.go

```diff
@@ -0,0 +1,45 @@
+package controller
+
+import "github.com/grafana/agent/component"
+
+// ComponentRegistry is a collection of registered components.
+type ComponentRegistry interface {
+	// Get looks up a component by name.
+	Get(name string) (component.Registration, bool)
+}
+
+// DefaultComponentRegistry is the default [ComponentRegistry] which gets
+// components registered to github.com/grafana/agent/component.
+type DefaultComponentRegistry struct{}
+
+// Get retrieves a component using [component.Get].
+func (reg DefaultComponentRegistry) Get(name string) (component.Registration, bool) {
+	return component.Get(name)
+}
+
+// RegistryMap is a map which implements [ComponentRegistry].
+type RegistryMap map[string]component.Registration
+
+// Get retrieves a component using [component.Get].
+func (m RegistryMap) Get(name string) (component.Registration, bool) {
+	reg, ok := m[name]
+	return reg, ok
+}
+
+// MultiRegistry returns a ComponentRegistry which combines the input
+// registries. Earlier registries take precednece over later registries if a
+// component is registered in both.
+func MultiRegistry(reg ...ComponentRegistry) ComponentRegistry {
```

Comment:
Whats the purpose of multi registries? Is this mostly for testing?

Reply:
> Whats the purpose of multi registries? Is this mostly for testing?

Yeah, I'm not using it right now so I could take it out, but it's intended for testing, where you want to overlay custom components on top of the default set without modifying the default set

Reply:
It feels like complexity we don't need right now. That makes me have to consider things like multi registration. Is it possible to make it an interface and then put multi capability around testcomponent?

Reply:
I'm not sure what you're suggesting. I'll just take the code out since it's not being called anywhere right now anyway.